utils/cogs/Administration.py

```diff
@@ -82,6 +82,75 @@ async def add_user(self, ctx, user: str = None, user_email: str = None, create_o
+        else:
+            users_to_prune = []
+            user_dict = utils.get_users(self.client, account_email)
+            print(user_dict)
+            for account_id in user_dict.keys():
+                print(utils.get_history(self.client, account_id, account_email))
+                if not utils.get_history(self.client, account_id, account_email):
+                    print("user has not watched anything")
+                    users_to_prune.append(user_dict.get(account_id))
+                    print(user_dict.get(account_id))
+
+
+            await ctx.send(embed=await embeds.user_list(users_to_prune, len(users_to_prune), account_email))
+            for user_email in users_to_prune:
+                if db_driver.check_user_email(self.client.db_cur, user_email):
+                    user = db_driver.grab_user(self.client.db_cur, user_email)
+                    for plex_server_account in self.client.plex_connections:
+                        if plex_server_account['account'].email == user[2]:
+                            # Remove User From Plex Share
+                            if plex.remove_user(plex_server_account['account'], user_email):
+                                if overseerr_enabled and user[5] != "None":
+                                    # Delete Overseerr Account
+                                    if overseerr.delete_user(overseerr_api, overseerr_server, user[5]):
+                                        await ctx.send(embed=await embeds.generic_success(
+                                            f"User with email: {user_email} successfully removed from Plex and Overseerr"))
+                                        db_driver.delete_user(self.client.db_cur, self.client.db_con, user_email)
+                                    else:
+                                        await ctx.send(embed=await embeds.generic_failure(
+                                            f"Successfully removed user with email: {user_email} from Plex share. However, Overseerr account deletion has failed. See logs for more info.\n\nUser has been removed from DB, please manually remove Overseerr account."))
+                                        db_driver.delete_user(self.client.db_cur, self.client.db_con, user_email)
+                                else:
+                                    await ctx.send(embed=await embeds.generic_success(
+                                        f"User with email: {user_email} successfully removed from Plex"))
+                                    db_driver.delete_user(self.client.db_cur, self.client.db_con, user_email)
+                            else:
+                                await ctx.send(embed=await embeds.generic_failure(
+                                    f"Could not remove {user_email} from Plex share. See container logs for more info."))
+                else:
+                    await ctx.send(embed=await embeds.generic_failure(f"email: {user_email} not found in DB"))
```

Comment:
Bro there is an entire generic method for doing this, but instead of using it, you copy pasted it

Reply:
might fix later